app/public/model/filtering/FilterModel.js

```diff
@@ -0,0 +1,121 @@
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { Observable } from '/js/src/index.js';
+
+/**
+ * Class for managing the filters on a general level
+ */
+export default class FilterModel extends Observable {
+    constructor() {
+        super();
+        this._activeFilters = {};
+    }
+
+    reset() {
+        this._activeFilters = {};
+        this.notify();
+    }
+
+    handleValueChange(field, type, value) {
+        switch (type) {
+            case 'match':
+            case 'exclude': {
+                this._activeFilters[field][type].push(value);
+                break;
+            }
+            case 'from':
+            case 'to': {
+                this._activeFilters[field][type] = [];
+                this._activeFilters[field][type].push(value);
+                break;
+            }
+            default: break;
+        }
+    }
+
+    resetFilterValue(field, type) {
+        this._activeFilters[field][type] = [];
+    }
+
+    prepareObjectStructure(field, type) {
+        if (! Object.prototype.hasOwnProperty.call(this._activeFilters, field)) {
+            this._activeFilters[field] = {
+                [type]: [],
+            };
+        }
+        if (! Object.prototype.hasOwnProperty.call(this._activeFilters[field], type)) {
+            this.resetFilterValue(field, type);
+        }
+    }
+
+    addFilter(field, value, type) {
+        this.prepareObjectStructure(field, type);
+        if (['from', 'to'].includes(type)) {
+            this.resetFilterValue(field, type);
+        }
+        this._activeFilters[field][type].push(value);
+        this.notify();
+    }
+
+    removeFilter(field, value, type) {
+        this._activeFilters[field][type] = this._activeFilters[field][type].filter((element) => element !== value);
+        this.notify();
+    }
+
+    filterTypesMapping(filterType, value) {
+        switch (filterType) {
+            case 'match': return `[or][like]=%${value}%`;
+            case 'exclude': return `[and][notLike]=%${value}%`;
+            case 'from': return `[gte]=${value}`;
+            case 'to': return `[lte]=${value}`;
+            default: return `=${value}`;
+        }
+    }
+
+    get activeFilters() {
+        return this._activeFilters;
+    }
+
+    get filterObjects() {
+        const result = Object.keys(this._activeFilters).map((field) =>
+            Object.keys(this._activeFilters[field]).reduce((typeAcc, currentType) => {
+                this._activeFilters[field][currentType].forEach((value) => typeAcc.push({
+                    field: field,
+                    type: currentType,
+                    value: value,
+                }));
+                return typeAcc;
+            }, []));
+        return result.flat();
```

Comment:
It can be much shorter, e.g.:
```
return Object.entries(this._activeFilters)
            .map(([field, typeToValue]) =>
                Object.entries(typeToValue)
                    .map(([type, value]) => ({ field, type, value })))
            .flat(2);
```

Reply:
Nice idea, but `value` is an array, so another iteration over it is needed. Anyway, it looks better this way.